### PR TITLE
added forms toggler

### PR DESCRIPTION
**Pull request by marat-sadriev**
# Summary

#### Screenshots in case of UI changes

Снимок экрана 2020-07-16 в 14 47 36

# Test plan

#### List of steps to manually test introduced functionality:

- Checkout to branch locally
- Go to /login page
- try to change between sign up / sign in forms

# Review notes

#### While reviewing pull-request (especially when it's your pull-request), please make sure that:

- you named PR close to issue name and linked that issue
- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### CODE REVIEW

File: components/organisms/SignUpForm/SignUpFormContent.js [hunk 1]
```diff
@@ -111,11 +111,37 @@ const SignUpFormContent = ({ onSubmit }) => {
...
+      },
+    ],
+    submit: values => onSubmit(values),
+  };
   return (
     <>
-      <StyledTitle>Create an account</StyledTitle>
-
-      <Form form={form} />
+      <StyledTitle>{activeForm === 'signUp' ? 'Create an account' : 'Log in'}</StyledTitle>
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpFormContent.js:143]: Can you compute this values (title, active form) outside from `return` render function? Render function will be a much clearer

File: components/organisms/SignUpForm/SignUpFormContent.js [hunk 2]
```diff
@@ -111,11 +111,37 @@ const SignUpFormContent = ({ onSubmit }) => {
     submit: values => onSubmit(values),
   };
 
+  const SignInForm = {
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpFormContent.js:114]: You can compute a form current fields using this recipe:

```
const isSignUp = activeForm === 'signUp';

const formFields = [
   isSignUp && nameFieldData,
   isSignUp && secondNameFieldData,
  emailFieldData,
  passwordFieldData,
  submitButtonData <- just change button text & name here,
].filter(Boolean);
```

File: components/organisms/SignUpForm/SignUpFormContent.js [hunk 3]
```diff
@@ -67,55 +67,71 @@ export const LinksContainer = styled.div`
...
+      name: 'signUp',
+      initialValue: 'Sign Up',
+    },
+    {
+      type: 'submit',
+      name: 'signIn',
+      initialValue: 'Sign In',
+    },
+  ];
+  const currentFormFields = [];
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpFormContent.js:126]: `const currentFormFields = {} `? and looks like submit property `submit: values => onSubmit(values)` should be here too

File: pages/logIn/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import SignUp from './logIn';
```
Comment (Nikita-Polyakov) [inline pages/logIn/index.js:1]: Rename to `LogIn` please

File: components/organisms/SignUpForm/SignUpForm.js [hunk 5]
```diff
@@ -1,16 +1,38 @@
...
-  return <SignUpFormContent error={error} loading={loading} onSubmit={signUp} />;
+  const [activeForm, toggleSignForm] = useState('signUp');
+  const handleActiveFormChoose = () => {
+    activeForm === 'signUp' ? toggleSignForm('signIn') : toggleSignForm('signUp');
+  };
+  const toggleButtonText = activeForm === 'signUp' ? 'Already signed up' : 'Sign up';
+  return (
+    <StyledFormWrapper>
+      <SignUpFormContent error={error} loading={loading} onSubmit={signUp} activeForm={activeForm} />
+      <StyledToggleForm onClick={() => handleActiveFormChoose()}>{toggleButtonText}</StyledToggleForm>
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpForm.js:33]: You can just pass a handler without wrapping in another arrow function `onClick={handleActiveFormChoose}`

File: components/organisms/SignUpForm/SignUpFormContent.js [hunk 6]
```diff
@@ -67,55 +67,71 @@ export const LinksContainer = styled.div`
...
+      initialValue: 'Sign Up',
+    },
+    {
+      type: 'submit',
+      name: 'signIn',
+      initialValue: 'Sign In',
+    },
+  ];
+  const currentFormFields = {
+    fields: allFields.filter(field => fieldNames[field.name]),
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpFormContent.js:127]: Can you define property submit here, in object declaration?

File: components/organisms/SignUpForm/SignUpForm.js [hunk 7]
```diff
@@ -1,16 +1,38 @@
...
   // todo remove callbacks from mutation hooks
   const {
     signUp,
     mutationResult: { error, loading },
   } = useSignUp({ onSuccess });
 
-  return <SignUpFormContent error={error} loading={loading} onSubmit={signUp} />;
+  const [activeForm, toggleSignForm] = useState('signUp');
+  const handleActiveFormChoose = () => {
+    activeForm === 'signUp' ? toggleSignForm('signIn') : toggleSignForm('signUp');
```
Comment (Nikita-Polyakov) [inline components/organisms/SignUpForm/SignUpForm.js:27]: Let's define `signUp` `signIn` strings as constants outside of declaration component:
```
const SIGN_IN_FORM = 'signIn';
const SIGN_UP_FORM = 'signUp';
```

Then import them and use in the related `SignUpFormContent`, to clarify dependencies between this components

File: components/organisms/SignUpForm/SignUpForm.js [hunk 8]
```diff
@@ -1,16 +1,38 @@
...
 import useSignUp from './useSignUp';
 
+export const StyledFormWrapper = styled.div`
+  text-align: center;
+`;
+
+export const StyledToggleForm = styled.div`
+  display: inline-block;
+  cursor: pointer;
+  color: #00f;
```
Comment (nadezhdakharchuk) [inline components/organisms/SignUpForm/SignUpForm.js:14]: Could you please move the color to theme variable

File: components/organisms/SignUpForm/SignUpFormContent.js (hunk 6 above)
Comment (nadezhdakharchuk) [inline components/organisms/SignUpForm/SignUpFormContent.js:127]: This line can be refactored like this:
`fields: allFields.filter(({ name }) => fieldNames[name]),`

File: components/organisms/SignUpForm/SignUpForm.js (hunk 7 above)
Comment (marat-sadriev) [reply components/organisms/SignUpForm/SignUpForm.js]: Do you say about relocate them in vars.js file?

**Pull request merged**